### ISSUE TITLE
Question on getSymbol function

### ISSUE THREAD

**Issue by bialikover** [ellunium/sketch-measure-latest#1]
@ellunium  I see that the `idx++` was removed in this commit:

https://github.com/ellunium/sketch-measure-latest/commit/b5518a2616bafe77db13ff7862d800cd20eb683d#diff-7fe6147dac26b2b69e1796156f5f9e9dL2667

And in the master branch `idx`  is still used to access the children elements:

```javascript
var symbolMasterObjectID = self.toJSString(symbolChildren[idx].objectID());
```

https://github.com/ellunium/sketch-measure-latest/blob/master/Sketch%20Measure.sketchplugin/Contents/Sketch/library/common.js#L2644

I was wondering why the `idx` was removed and if its ok to assume that `idx` will always be `0`...

**Comment by ellunium** [comment]
@bialikover, it was removed as temporary fix to export files without symbols, since sketch 53.1 had a bug that prevented going through the symbol master children properly.

With Sketch v53.2 (72643) this has been fixed and idx was back in the picture.

Idx will always start at zero, but by the end of the while loop it is incrementes by 1.

It’s there to loop through the symbol master children and make sure we get the right layers and overrides when exporting.

You can see the whole discussion that took place to get to the latest version here:
https://github.com/utom/sketch-measure/issues/536

I hope that helps. :)

**Comment by bialikover** [comment]
Hi @ellunium Thanks a lot for clarifying. 

I had this doubt because in the master branch `idx` is not incrementing at the end of the while loop :(

https://github.com/ellunium/sketch-measure-latest/blob/master/Sketch%20Measure.sketchplugin/Contents/Sketch/library/common.js#L2666

**Comment by ellunium** [comment]
@bialikover nice catch. 

To be fair, I thought I had added `idx++` back there 😅😂, oh well.

I've spent days trying to figure out what was wrong there until I found out that `layer.symbolMaster().children()` was returning different children (sometimes) in `Sketch v52` and `Sketch v53`.

Whenever I try to fix anything related to `Measure` I try to keep as close as possible to @utom's master branch code. Then my first thought was to align the output of `symbolChildren` to be the same on both `Sketch v52` and `Sketch v53`, so that I could leave the rest of the function alone and working as before,  although I came close, it was not flexible or scalable enough to handle all cases and in the cases where my function aligned the output, sometimes it would randomly not work still.

While I was creating this normalizing function to align the output of `symbolChildren`, v53.2 came out and in one of my tests it worked properly, but I guess this was one of the times where I just decided to reset my progress and start from scratch again.

I do this often by the way, I just delete all my debugging files, code, etc; and start the debugging process from scratch again, because sometimes it feels like I'm zooming in too much on one possible problem/solution and I start to feel like I'm missing the big picture and probably missing other opportunities to fix that given problem and other solutions as well.

With `idx = 0`, I think what's happening then is that we always get the `MSSymbolMaster` instance on Sketch v53.2 for any given symbol instance, **I guess** that's enough to make the code work properly as is(???). It seems so.

One thing you will notice about me is that, I don't really have all the answers nor I can code that well, and sometimes things  just work 😂.

I hope this clarifies a little better. 
Sorry for the misleading answers above ;)
